hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -39,31 +38,76 @@ func New(capiProviderImage string) *OpenStack {
 
 func (a OpenStack) ReconcileCAPIInfraCR(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster,
 	controlPlaneNamespace string, apiEndpoint hyperv1.APIEndpoint) (client.Object, error) {
-	openstackCluster := &capiopenstack.OpenStackCluster{
+	openStackCluster := &capo.OpenStackCluster{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      hcluster.Name,
 			Namespace: controlPlaneNamespace,
 		},
 	}
-	openstackPlatform := hcluster.Spec.Platform.OpenStack
-	if openstackPlatform != nil {
-		openstackCluster.Spec.IdentityRef = capiopenstack.OpenStackIdentityReference{
-			Name:      openstackPlatform.CloudsYamlSecret.Name,
+	openStackPlatform := hcluster.Spec.Platform.OpenStack
+	if openStackPlatform != nil {
```

Comment:
This logic is weird. We should flip this to `if openStackPlatform == nil` and return an error.

Reply:
done